src/Microsoft.Health.Fhir.Api/Features/Routing/UrlResolver.cs

```diff
@@ -168,10 +169,40 @@ public Uri ResolveRouteUrl(IEnumerable<Tuple<string, string>> unsupportedSearchP
 
                     if (string.Equals(searchParam.Key, KnownQueryParameterNames.Sort, StringComparison.OrdinalIgnoreCase))
                     {
-                        if (resultSortOrder?.Count > 0)
+                        switch (resultSortOrder?.Count)
                         {
+                            case null:
+                            case 0:
+                                break;
+
                             // rewrite the sort order based on the sort order that was actually applied
-                            routeValues.Add(searchParam.Key, new StringValues(resultSortOrder.Select(s => $"{(s.sortOrder == SortOrder.Ascending ? string.Empty : "-")}{s.searchParameterInfo.Code}").ToArray()));
+                            case 1 when resultSortOrder[0].sortOrder == SortOrder.Ascending:
+                                routeValues.Add(searchParam.Key, resultSortOrder[0].searchParameterInfo.Name);
+                                break;
+                            default:
+                                var sb = new StringBuilder();
+                                bool first = true;
+                                foreach ((SearchParameterInfo searchParameterInfo, SortOrder sortOrder) in resultSortOrder)
+                                {
+                                    if (first)
+                                    {
+                                        first = false;
+                                    }
+                                    else
+                                    {
+                                        sb.Append(',');
+                                    }
+
+                                    if (sortOrder == SortOrder.Descending)
+                                    {
+                                        sb.Append('-');
+                                    }
+
+                                    sb.Append(searchParameterInfo.Name);
+                                }
+
+                                routeValues.Add(searchParam.Key, sb.ToString());
```

Comment:
>routeValues [](http://example.com/codeflow?start=32&length=11)

```
routeValues.Add(searchParam.Key, string.Join(',' resultSortOrder.Select(s => $"{(s.sortOrder == SortOrder.Ascending ? string.Empty : "-")}{s.searchParameterInfo.Code}")
```
to replace previous code, this one looks a bit chunky.

Reply:
Yeah. True.